res/css/views/right_panel/_TimelineCard.scss

```diff
@@ -110,6 +116,14 @@ limitations under the License.
             }
         }
 
+        &[data-layout=group] {
+            // Read receipt group on compact modern layout
+            // mx_TimelineCard is wrapped by mx_MatrixChat_useCompactLayout
+            .mx_MatrixChat_useCompactLayout & .mx_ReadReceiptGroup {
+                top: var(--TimelineCard_ReadReceiptGroup-inset-block-start);
```

Comment:
It's not entirely clear why this is necessary? Is this overriding something?

Reply:
Of course, this is necessary because of [this line](https://github.com/matrix-org/matrix-react-sdk/blame/fee6a59734477cc07ab9064d98a7542bdb9a1261/res/css/views/rooms/_EventTile.scss#L1156). Otherwise it would not work because  `mx_TimelineCard` is a child element of `.mx_MatrixChat_useCompactLayout`, as the comment says.

|Before|After|
|---------|------|
|![before3](https://user-images.githubusercontent.com/3362943/177003579-f645d526-3504-4adb-9b85-09274655bc7c.png)|![after3](https://user-images.githubusercontent.com/3362943/177003405-527c595c-614a-4f46-9931-ffb2da333248.png)|
||↑ `-2rem` is canceled|

I understand it is hard to see why that is necessary- that is how much the current style rule structure is complex... :disappointed:

Reply:
Ah, I see - could you please update the comment to be a little clearer about this?